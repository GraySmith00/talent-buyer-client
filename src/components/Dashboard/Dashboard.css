.dashboard {
  display: flex;
  background-color: #21222b;
  color: #fff;
  overflow: scroll; }
  .dashboard .main-content {
    padding: 3rem 5rem;
    width: 100%; }
  .dashboard .dates {
    display: flex; }
  .dashboard .calendar {
    height: 320px;
    width: 420px; }
  .dashboard .offers {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-left: 3rem;
    width: 100%; }
  .dashboard h2 {
    margin-top: 0; }
  .dashboard .create-offer-button {
    display: flex;
    align-items: center;
    height: 3rem;
    width: 11rem;
    padding: 13px;
    border: none;
    border-radius: 5px;
    font-size: 1.2rem;
    color: #fff;
    cursor: pointer;
    -webkit-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
    box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75); }
  .dashboard .fas {
    margin-right: 1rem; }
  .dashboard .view-offer-button {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 2rem;
    width: 5rem;
    border: none;
    border-radius: 5px;
    font-size: 0.7rem;
    background-color: #01bbd3;
    color: #fff;
    cursor: pointer;
    -webkit-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
    box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75); }
  .dashboard .recents {
    display: grid;
    grid-template-columns: repeat(2, 10fr);
    grid-gap: 2rem;
    margin-top: 2rem; }
    .dashboard .recents h2 {
      margin-bottom: 0; }
