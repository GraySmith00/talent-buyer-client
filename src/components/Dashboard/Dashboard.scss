.dashboard {
  display: flex;
  background-color: #21222b;
  color: #fff;
  overflow: scroll;

  .main-content {
    padding: 3rem 5rem;
    width: 100%;
  }

  .dates {
    display: flex;
  }

  .calendar {
    height: 320px;
    width: 420px;
  }

  .offers {
    display: flex;
    flex-direction: column;
    margin-left: 3rem;
    width: 100%;
  }

  h2 {
    margin-top: 0;
  }

  .create-offer-button {
    display: flex;
    align-items: center;

    height: 3rem;
    width: 11rem;
    padding: 13px;
    border: none;
    border-radius: 5px;
    font-size: 1.2rem;
    color: #fff;
    cursor: pointer;

    -webkit-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
    box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
  }

  .fas {
    margin-right: 1rem;
  }

  .table-headings {
    display: grid;
    grid-template-columns: repeat(4, 10fr);
    border-bottom: 1px solid #fff;
    align-items: center;
    padding-left: 2rem;
    margin: 1rem 0 0 0;
    h3 {
      margin-bottom: 5px;
    }
  }

  .view-offer-button {
    display: flex;
    align-items: center;
    justify-content: center;

    height: 2rem;
    width: 5rem;

    border: none;
    border-radius: 5px;
    font-size: 0.7rem;
    background-color: #01bbd3;
    color: #fff;
    cursor: pointer;

    -webkit-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
    box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
  }

  .recents {
    display: grid;
    grid-template-columns: repeat(2, 10fr);
    grid-gap: 2rem;
    margin-top: 2rem;

    h2 {
      margin-bottom: 0;
    }
  }
}
