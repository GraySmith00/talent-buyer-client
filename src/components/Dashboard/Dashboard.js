import React, { Component } from 'react';
import Calendar from 'react-calendar';
import axios from 'axios';

import Nav from '../Nav/Nav';
import OfferModal from '../OfferModal/OfferModal';
import ModalButton from '../styledComponents/ModalButton';

import './Dashboard.css';
import OffersContainer from '../OffersContainer/OffersContainer';
import EditOfferModal from '../EditOfferModal/EditOfferModal';
import RecentOffers from '../RecentOffers/RecentOffers';
import Watchlist from '../Watchlist/Watchlist';
import ViewOfferModal from '../ViewOfferModal/ViewOfferModal';

class Dashboard extends Component {
  state = {
    date: new Date(),
    offerModalOpen: false,
    editOfferModalOpen: false,
    viewOfferModalOpen: false,
    editOfferId: '',
    viewOfferId: ''
  };

  componentDidMount() {
    axios.defaults.headers.common['Authorization'] = localStorage.getItem(
      'jwtToken'
    );
  }

  handleDateChange = date => this.setState({ date });
  openOfferModal = () => this.setState({ offerModalOpen: true });
  closeOfferModal = () => this.setState({ offerModalOpen: false });
  openEditOfferModal = id => {
    this.setState({ editOfferModalOpen: true, editOfferId: id });
  };
  closeEditOfferModal = () => this.setState({ editOfferModalOpen: false });
  openViewOfferModal = id => {
    this.setState({ viewOfferModalOpen: true, viewOfferId: id });
  };
  closeViewOfferModal = () => this.setState({ viewOfferModalOpen: false });

  render() {
    const {
      offerModalOpen,
      date,
      editOfferId,
      editOfferModalOpen,
      viewOfferId,
      viewOfferModalOpen
    } = this.state;
    const dateString = date.toString().slice(0, 15);

    return (
      <div className="dashboard">
        {offerModalOpen && (
          <OfferModal closeOfferModal={this.closeOfferModal} date={date} />
        )}
        {editOfferModalOpen && (
          <EditOfferModal
            offerId={editOfferId}
            closeEditOfferModal={this.closeEditOfferModal}
          />
        )}
        {viewOfferModalOpen && (
          <ViewOfferModal
            offerId={viewOfferId}
            closeViewOfferModal={this.closeViewOfferModal}
          />
        )}
        <Nav />
        <div className="main-content">
          <div className="dates">
            <Calendar
              onChange={this.handleDateChange}
              value={this.state.date}
              className="calendar"
            />
            <div className="offers">
              <h2>{dateString}</h2>
              <ModalButton
                className="create-offer-button"
                style={{ marginTop: '0' }}
                onClick={this.openOfferModal}
              >
                <i className="fas fa-plus" />
                Create Offer
              </ModalButton>
              <div className="table-headings">
                <h3>Artist</h3>
                <h3>Status</h3>
                <h3>Edit</h3>
                <h3>View</h3>
              </div>
              <OffersContainer
                date={date}
                openEditOfferModal={this.openEditOfferModal}
                openViewOfferModal={this.openViewOfferModal}
              />
            </div>
          </div>
          <div className="recents">
            <Watchlist />
            <RecentOffers openEditOfferModal={this.openEditOfferModal} />
          </div>
        </div>
      </div>
    );
  }
}

export default Dashboard;
