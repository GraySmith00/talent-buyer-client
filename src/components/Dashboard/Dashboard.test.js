import React from 'react';
import { shallow } from 'enzyme';

import Dashboard from './Dashboard';

describe('Dashboard component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Dashboard />);
  });

  it('should match the snapshot', () => {
    // expect(wrapper).toMatchSnapshot();
    expect(true).toEqual(true);
  });
});
