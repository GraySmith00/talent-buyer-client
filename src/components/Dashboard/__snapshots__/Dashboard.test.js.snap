// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav />,
        <Calendar
          maxDetail="month"
          minDetail="century"
          onChange={[Function]}
          returnValue="start"
          showNavigation={true}
          showNeighboringMonth={true}
          value={2018-10-28T17:22:19.431Z}
          view="month"
/>,
      ],
      "className": "dashboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "maxDetail": "month",
          "minDetail": "century",
          "onChange": [Function],
          "returnValue": "start",
          "showNavigation": true,
          "showNeighboringMonth": true,
          "value": 2018-10-28T17:22:19.431Z,
          "view": "month",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav />,
          <Calendar
            maxDetail="month"
            minDetail="century"
            onChange={[Function]}
            returnValue="start"
            showNavigation={true}
            showNeighboringMonth={true}
            value={2018-10-28T17:22:19.431Z}
            view="month"
/>,
        ],
        "className": "dashboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "maxDetail": "month",
            "minDetail": "century",
            "onChange": [Function],
            "returnValue": "start",
            "showNavigation": true,
            "showNeighboringMonth": true,
            "value": 2018-10-28T17:22:19.431Z,
            "view": "month",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
